### PR TITLE
COR-404 foregin key error translation

### PR DESCRIPTION
**Pull request by morus12**

### CODE REVIEW

File: database/errors_test.go [hunk 1]
```diff
@@ -252,14 +252,20 @@ func TestDatabaseToApiError(t *testing.T) {
 			err:      &ConnectionError{},
 			expected: "rpc error: code = Internal desc = [GAD.DAT.262] internal system error encountered",
 		},
+		{
+			name:     "foregin key",
```
Comment (apynes) [inline database/errors_test.go:256]:
```suggestion
			name:     "foreign key",
```

**Review by apynes — APPROVED** [review]
typo, other than that it looks good

File: database/errors.go [hunk 2]
```diff
@@ -315,6 +315,9 @@ func DatabaseToApiError(primary qb.Table, dbError error) error {
 	case *UniqueConstraintError:
 		err = status.Error(codes.InvalidArgument, fmt.Sprintf("%s %s unique constraint violation: %s",
 			prefix, primary.GetName(), dbError))
+	case *InvalidForeignKeyError:
+		err = status.Error(codes.InvalidArgument, fmt.Sprintf("%s %s foregin key violation: %s",
```
Comment (nmarie22) [inline database/errors.go:319]: Same typo here

**Review by nmarie22 — CHANGES_REQUESTED** [review]

**Review by nmarie22 — APPROVED** [review]

**Comment by nmarie22** [comment]
@morus12 change is ready to be released and version updated on core and connect

**Pull request merged**